builder/vmware/common/driver_esxi.go

```diff
@@ -747,12 +748,18 @@ func (d *EsxiDriver) connect() error {
 		auth = append(auth, gossh.PublicKeys(signer))
 	}
 
+	knownHostsFile := filepath.Join(os.Getenv("HOME"), ".ssh", "known_hosts")
+	hostKeyCallback, err := knownhosts.New(knownHostsFile)
+	if err != nil {
+		return err
+	}
+
 	sshConfig := &ssh.Config{
 		Connection: ssh.ConnectFunc("tcp", address),
 		SSHConfig: &gossh.ClientConfig{
 			User:            d.Username,
 			Auth:            auth,
-			HostKeyCallback: gossh.InsecureIgnoreHostKey(),
+			HostKeyCallback: hostKeyCallback,
```

Comment:
That I wonder, since the machine we're connecting to is a temporary VM, we probably don't have an entry for it in the local `known_hosts`, won't that prevent us from connecting later?